tests/data/test_dataset.py

```diff
@@ -0,0 +1,78 @@
+# Copyright 2018-2022 Xanadu Quantum Technologies Inc.
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+#     http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""
+Unit tests for the :class:`pennylane.data.Dataset` class and its functions.
+"""
+import pennylane as qml
+import dill
+import zstd
+from unittest.mock import patch, mock_open
+
+
+def test_build_dataset():
+    """Test that a dataset builds correctly and returns the correct values."""
+    hamiltonian = qml.Hamiltonian(coeffs=[1], observables=[qml.PauliZ(wires=0)])
+    test_dataset = qml.data.Dataset(kw1=1, kw2="2", kw3=[3], hamiltonian=hamiltonian)
+
+    assert test_dataset.kw1 == 1
+    assert test_dataset.kw2 == "2"
+    assert test_dataset.kw3 == [3]
+    assert test_dataset.hamiltonian == hamiltonian
+
+
+def test_write_dataset():
+    """Test that datasets are saved correctly."""
+    open_mock = mock_open()
+    test_dataset = qml.data.Dataset(kw1=1, kw2="2", kw3=[3])
+
+    with patch("builtins.open", open_mock):
```

Comment:
Any reason for mocking here rather than using a pytest fixture? https://docs.pytest.org/en/6.2.x/fixture.html

They have quite a good selection of fixtures, including `tmpdir`, `tmp_path`, `testdir`, etc.

Reply:
We used the `tmp_path` fixture in a previous commit but I thought it was not recommended to write to disk/storage during tests as it is slower. If it's not a concern for these small files, I can change it back.